mesonbuild/dependencies/cmake.py

```diff
@@ -125,9 +125,7 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             raise self._gen_exception('Unable to obtain CMake system information')
         self.cmakeinfo = cmakeinfo
 
-        package_version = kwargs.get('cmake_package_version', '')
-        if not isinstance(package_version, str):
-            raise DependencyException('Keyword "cmake_package_version" must be a string.')
+        package_version = T.cast('str', kwargs.get('cmake_package_version', ''))
```

Comment:
Is the cast really needed here? The typed dict has the right type and the `.get()` default too, thus it should not be needed, or is `mypy` not smart enough? From the look of it I would also say that the `.get()` is not necessary as the argument is declared with a default, but I don't really know how that part of the machinery works...

Reply:
We aren't passing the typed dict through. I attempted to do that originally, and discovered we have paths that hit this that don't have the full set of kwargs passed in. I removed the `isinstance()` check sine that was really for validating code coming through the interpreter, but the cast needed to remain otherwise mypy interprets the type as `str | Any`

Reply:
I see. I didn't pay attention. Thanks for the explanation.